pkg/storaged/content-views.jsx

```diff
@@ -0,0 +1,744 @@
+        } else if (block_part) {
+            name = utils.block_name(block);
+            danger = _("Deleting a partition will delete all data in it.");
+        }
+
+        if (name) {
+            dialog.open({ Title: cockpit.format(_("Please confirm deletion of $0"), name),
+                          Alerts: utils.get_usage_alerts(client, target.path),
+                          Fields: [
+                          ],
+                          Action: {
+                              Danger: danger,
+                              Title: _("Delete"),
+                              action: function () {
+                                  if (lvol)
+                                      return lvol.Delete({ 'tear-down': { t: 'b', v: true }
+                                      });
+                                  else if (block_part)
+                                      return block_part.Delete({ 'tear-down': { t: 'b', v: true }
+                                      });
+                              }
+                          }
+            });
+        }
+    }
+
+    if (is_partition || lvol) {
+        var excuse = null;
+        if (client.is_old_udisks2 && is_crypto && client.blocks_cleartext[block.path])
+            excuse = _("Can't delete while unlocked");
+        add_action(_("Delete"), delete_, excuse);
+    }
+
+    return {
+        renderers: tabs,
+        actions: [ <div>{tab_actions}</div> ],
+        row_action: row_action,
+    };
+}
+
+function block_description (client, block) {
+    var usage;
+    var block_pvol = client.blocks_pvol[block.path];
+
+    if (block.IdUsage == "filesystem") {
+        usage = cockpit.format(C_("storage-id-desc", "$0 File System"), block.IdType);
+    } else if (block.IdUsage == "raid") {
+        if (block_pvol && client.vgroups[block_pvol.VolumeGroup]) {
+            var vgroup = client.vgroups[block_pvol.VolumeGroup];
+            usage = cockpit.format(_("Physical volume of $0"), vgroup.Name);
+        } else if (client.mdraids[block.MDRaidMember]) {
+            var mdraid = client.mdraids[block.MDRaidMember];
+            usage = cockpit.format(_("Member of RAID Device $0"), utils.mdraid_name(mdraid));
+        } else if (block.IdType == "LVM2_member") {
+            usage = _("Physical Volume");
+        } else {
+            usage = _("Member of RAID Device");
+        }
+    } else if (block.IdUsage == "crypto") {
+        usage = C_("storage-id-desc", "Encrypted data");
+    } else if (block.IdUsage == "other") {
+        if (block.IdType == "swap") {
+            usage = C_("storage-id-desc", "Swap Space");
+        } else {
+            usage = C_("storage-id-desc", "Other Data");
+        }
+    } else {
+        usage = C_("storage-id-desc", "Unrecognized Data");
+    }
+
+    return {
+        size: utils.fmt_size(block.Size),
+        text: usage
+    };
+}
+
+function append_row (rows, level, key, name, desc, tabs, job_object) {
```

Comment:
A size of 0 doesn't have a unit, which looks a bit strange. On master, there is no `0`. I suggest writing `0 B` or `empty`.

![screenshot from 2016-11-18 17-14-12](https://cloud.githubusercontent.com/assets/8711649/20437294/bcbdaa3c-adb3-11e6-897c-27b913bc2ea0.png)